### ISSUE TITLE
Error adding Custom Domain to CDN

### ISSUE THREAD

**Issue by TheRSP** [CompositionalIT/farmer#441]
The following Farmer program fails to deploy with the following error. I'm not sure what to do about this. If I remove the domain name the problem goes away.

```
Deploying ARM template (please be patient, this can take a while)...
Unhandled exception. System.Exception: Azure Error: InvalidTemplate
Message: Deployment template validation failed: 'The template resource 'my-7e453-cdn-endpoint/my-7e453-cdn-endpointdomain' for type 'Microsoft.Cdn/profiles/endpoints/customDomains' at line '69' and column '62' has incorrect segment lengths. A nested resource type must have identical number of segments as its resource name. A root resource type must have segment length one greater than its resource name. Please see https://aka.ms/arm-template/#resources for usage details.'.
```

```f#
  let storageAcc = storageAccount {
    name "my7e453storage"
    sku Storage.Sku.Standard_LRS
    
    use_static_website "/path/to/content" "index.html"
  }

  let cdnProfile = cdn {
    name ("my7e453-cdn")
    add_endpoints [
      (endpoint {
        name ("my-7e453-cdn-endpoint")
        origin (storageAcc.WebsitePrimaryEndpoint |> System.Uri).Host
        optimise_for Cdn.OptimizationType.GeneralWebDelivery
        enable_https
        disable_http
        custom_domain_name "https://sub.example.com"
      })
    ]
  }

  // Generate the ARM template
  arm { 
    location Location.UKSouth
    add_resource storageAcc
    add_resource cdnProfile
  }
```

**Comment by isaacabraham** [comment]
@TheRSP hmmmm, that's not great. There's actually a branch that's reworking how we store and generate these paths which will help reduce the chances of this happening. In the meantime, could you also attach the generated ARM template here (it'll be in the `.farmer` folder).

I'm sure we can push out a fix for this relatively quickly.

**Comment by isaacabraham** [comment]
@TheRSP which version of Farmer are you on? The above code actually isn't valid in 1.2 since we changed the WebsitePrimaryEndpoint to be an expression.

**Comment by isaacabraham** [comment]
@TheRSP I've created a fix for this issue - Farmer was incorrectly emitting the custom domain section of the ARM template, which I have now fixed.

This will go out as part of 1.3.0-beta which will drop this week. Note that there is a small change you will need to do in order to upgrade: Instead of creating a URI and then getting the host for the storage account:

```fsharp
origin (storageAcc.WebsitePrimaryEndpoint |> System.Uri).Host
```

you will simply need to pass in the new `EndpointHost` member:

```fsharp
origin storageAcc.WebsitePrimaryEndpointHost
```

Basically the way we were doing it before (with a raw string) was fragile and not guaranteed to work depending on what URI the storage account created; the new approach uses an ARM expression instead and so should always match the generated web endpoint.

**Comment by TheRSP** [comment]
Sorry it took me so long to respond. The stated change is much nicer.
Thank you

**Comment by isaacabraham** [comment]
No worries. Thanks for spotting the issue!

**Comment by isaacabraham** [comment]
BTW if in the short term you want to fix the ARM templates generated, it's just going to be the Custom Domain name in the template - you'll need to prefix the path with the profile name e.g. in the example above, ""my7e453-cdn/"
